src/NuGet.Core/NuGet.Protocol/NameSpace/PackageSourceSection.cs

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace NuGet.Protocol
+{
+    public class PackageSourceSection
```

Comment:
Seems like this could be an internal type. 

Consider moving all this logic to the same assembly as the packagenamespacesconfiguration

Reply:
Made it internal type.